### ISSUE TITLE
Why are only a subset of species DB prefixes mentioned in the metadata/extensions-constraints.yaml ?

### ISSUE THREAD

**Issue by ValWood** [geneontology/go-site#1756]
https://github.com/geneontology/go-site/blob/master/metadata/extensions-constraints.yaml

**Comment by ValWood** [comment]
e.g.
- relation: has_input
  namespaces:
    - MGI
    - WB
    - PR
    - UniProtKB
    - CHEBI
    - PR
    - SO
    - CL
    - GO
    - WBbt



Pombase uses has_input 1000,s of times but does not seem to haveany violations. Is this set of extension constraints active?

(we were looking on the GO website for  the list of allowed extensions)

**Comment by pgaudet** [comment]
As far as I know nothing is checked in extensions 
@kltm

**Comment by kltm** [comment]
extensions-constraints.yaml is being used by @dustine32 for the GO-CAM import process and possibly other non-mainline tasks.

**Comment by pgaudet** [comment]
@kltm you mean this file is only used for imports?

**Comment by kltm** [comment]
My recollection is that this was something created while generalizing something found in the import code into ontobio. I do not believe it is broadly used outside of the original use case. This was wall some time ago. For more detail, you'd have have to ping @dustine32 .

**Comment by dustine32** [comment]
@kltm Correct, this is only ever used for validating extensions on annotations being imported to GO-CAM. Since this was developed while working with WB and MGI's annotations, we were only including prefixes used in those datasets. As we work with importing more datasets to GO-CAM we can add the necessary prefixes.

**Comment by ValWood** [comment]
Wouldn't it be better to have global annotation extension meta-data constraints, rather than Noctua specific ones?

**Comment by kltm** [comment]
@ValWood To this point, the only use and development time was for this specific use case. It may in fact expand into something more global in the future, but for the time being, the current actual use is limited.

**Comment by ValWood** [comment]
Is there a ticket anywhere to check  the syntax of the extensions?
If so, this can close.

**Comment by pgaudet** [comment]
No, there isn't. I'll change the title of the current ticket.

**Comment by pgaudet** [comment]
GORULE-0000001 is checking that namespaces area valid in extensions, see [GAF tests](https://github.com/geneontology/go-site/blob/master/docs/gorules_test_errors.gaf)

> ! FAILS GORULE:0000001 - TEST 2 - (see also GORULE:0000058) bad extension syntax															
PomBase	SPAC25B8.17	ypf1	is_active_in	GO:0005634	GO_REF:0000024	ISO	SGD:S000001583	C	GORULE_TEST:0000001-1 intramembrane aspartyl protease of the perinuclear ER membrane Ypf1 (predicted)	ppp81	protein	taxon:4896	20150305	PomBase	part_of
! FAILS GORULE:0000001 - (see also GORULE:0000058) bad extension filler: not a CURIE															
PomBase	SPAC25B8.17	ypf1	is_active_in	GO:0005634	GO_REF:0000024	ISO	SGD:S000001583	C	GORULE_TEST:0000001-2 intramembrane aspartyl protease of the perinuclear ER membrane Ypf1 (predicted)	ppp81	protein	taxon:4896	20150305	PomBase	part_of(x)
! FAILS GORULE:0000001 - TEST 3 - (see also GORULE:0000058) bad extension filler: space in ID															
PomBase	SPAC25B8.17	ypf1	is_active_in	GO:0005634	GO_REF:0000024	ISO	SGD:S000001583	C	GORULE_TEST:0000001-3 intramembrane aspartyl protease of the perinuclear ER membrane Ypf1 (predicted)	ppp81	protein	taxon:4896	20150305	PomBase	part_of(X:9 X:8)
! FAILS GORULE:0000001 - TEST 4 - (see also GORULE:0000058) bad extension filler: space in ID, with an additional valid expression
